### PR TITLE
Bindings refactoring & Automap bindings

### PR DESCRIPTION
**Pull request by Ch0wW**
Refactored the bindings, so that all new binding commands won't need to be copy/pasted.
Also added independant automap bindings, and added a menu control dedicated to it.

### CODE REVIEW

**Comment by AlexMax** [comment]
Nice work.  My only concern is that if you upgrade from an older version of Odamex and have an existing config, none of the automap keys will be bound by default.

**Comment by Ch0wW** [comment]
The last commit should rectify the issue you mentionned.

File: client/src/c_bind.cpp [hunk 1]
```diff
@@ -570,4 +549,94 @@ std::string C_GetKeyStringsFromCommand(const char *cmd, bool bTwoEntries)
...
+}
+END_COMMAND(bind)
+
+BEGIN_COMMAND(unbind)
+{
+	if (argc > 1) {
+		Bindings.UnbindKey(argv[1]);
+	}
+}
+END_COMMAND(unbind)
```
Comment (AlexMax) [inline client/src/c_bind.cpp:580]: Could you add a short little help if argc < 2 instead of printing nothing?

File: client/src/c_bind.cpp [hunk 2]
```diff
@@ -570,4 +549,94 @@ std::string C_GetKeyStringsFromCommand(const char *cmd, bool bTwoEntries)
...
+}
+END_COMMAND(doublebind)
+
+BEGIN_COMMAND(undoublebind)
+{
+	if (argc > 1) {
+		DoubleBindings.UnbindKey(argv[1]);
+	}
+}
+END_COMMAND(undoublebind)
```
Comment (AlexMax) [inline client/src/c_bind.cpp:596]: Could you add a short little help if argc < 2 instead of printing nothing?

File: client/src/c_bind.cpp [hunk 3]
```diff
@@ -570,4 +549,94 @@ std::string C_GetKeyStringsFromCommand(const char *cmd, bool bTwoEntries)
...
+}
+END_COMMAND(ambind)
+
+BEGIN_COMMAND(amunbind)
+{
+	if (argc > 1) {
+		AutomapBindings.UnbindKey(argv[1]);
+	}
+}
+END_COMMAND(amunbind)
```
Comment (AlexMax) [inline client/src/c_bind.cpp:611]: Could you add a short little help if argc < 2 instead of printing nothing?

**Review by leximayfield — CHANGES_REQUESTED** [review]
Mostly looks good, just have a few small things to take care of and I'll merge ASAP.

**Review by leximayfield — APPROVED** [review]

**Comment by AlexMax** [comment]
LGTM

**Pull request merged**